src/app/tab2/tab2.page.ts

```diff
@@ -22,6 +22,7 @@ export class Tab2Page {
   updateDisplayValue(displayValue: string) {
     this.displayValue += displayValue;
     this.calculatorService.setCurrentValue(this.displayValue);
+    this.displayValue = this.calculatorService.getCurrentValue();
```

Comment:
What would happends if change `this.display` value to getter `get displayValue (){return this.calculatorService.getCurrentValue()}`